ghcide/src/Development/IDE/Spans/Common.hs

```diff
@@ -76,13 +78,24 @@ data SpanDocUris =
 emptySpanDoc :: SpanDoc
 emptySpanDoc = SpanDocText [] (SpanDocUris Nothing Nothing)
 
+-- | Convert `SpanDoc` to Markdown format.
+--
+-- Return a list `Text` includes haddock, document uri and source code uri,
+-- each item can be empty and must end with '\\n' if exist.
```

Comment:
It would be good to explain why each item must end with a newline.

Also, I don't think all the items end with a newline - the ones produced by `spanDocUrisToMarkdown` don't, do they ?